lib/archethic/transaction_chain/transaction/data.ex

```diff
@@ -126,9 +162,20 @@ defmodule Archethic.TransactionChain.TransactionData do
 
   @spec cast(map()) :: t()
   def cast(data = %{}) do
+    code = Map.get(data, :code, "")
+
+    # our database contains both compressed and not compressed code
```

Comment:
This is not true since `cast` function is also used by API which will send uncompressed code